src/azul/service/async_manifest_service.py

```diff
@@ -63,40 +75,93 @@ class AsyncManifestService(AbstractService):
     """
     step_function_helper = StepFunctionHelper()
 
+    step_function_execution_retries = 2
+
     def __init__(self, state_machine_name):
         self.state_machine_name = state_machine_name
 
-    def start_generation(self, input: JSON) -> Token:
-        execution_id = str(uuid.uuid4())
-        self.step_function_helper.start_execution(self.state_machine_name,
-                                                  execution_id,
-                                                  execution_input=input)
-        return Token(execution_id=execution_id,
-                     request_index=0,
-                     wait_time=self._get_next_wait_time(0))
-
-    def inspect_generation(self, token) -> Union[Token, JSON]:
+    def start_generation(self,
+                         input: JSON,
+                         token: Optional[Token] = None,
+                         execution_id: Optional[str] = None,
+                         retry_index: int = 0) -> Token:
+        if execution_id is None:
+            execution_id = input['object_key'].rpartition('/')[2]
+            execution_id = execution_id.partition('.')[0]
+            assert '.' not in execution_id, execution_id
```

Comment:
If token is passed, this is redundant. That's inefficient and makes the code harder to understand. If the token and execution_id kwargs are mutually exclusive, an assertion should clarify that.

The fact that this and start_generation are mutually recursive pollutes the signature of this method. Even though clients only have to specify the positional argument, readers might be confused/scared by the complicated signature.